railties/lib/rails/commands/server/server_command.rb

```diff
@@ -132,27 +155,39 @@ class ServerCommand < Base # :nodoc:
       def initialize(args = [], local_options = {}, config = {})
         @original_options = local_options
         super
-        @server = self.args.shift
         @log_stdout = options[:daemon].blank? && (options[:environment] || Rails.env) == "development"
       end
 
       def perform
         set_application_directory!
         prepare_restart
-        Rails::Server.new(server_options).tap do |server|
-          # Require application after server sets environment to propagate
-          # the --environment option.
-          require APP_PATH
-          Dir.chdir(Rails.application.root)
+
+        # Require application after server sets environment to propagate
+        # the --environment option.
+        require APP_PATH
+        Dir.chdir(Rails.application.root)
+
+        server = Rails::Server.new(server_options)
+        handler = begin
+          server.handler
+        rescue Rails::Server::MissingError
+          puts HandlerSuggestion.message(using)
+          return
+        end
+
+        begin
```

Comment:
Same here. The purpose of the `Exiting` message is to show up, for example, when you `CTRL-C`/terminate a running server.